azure-pipelines-ci.yml

```diff
@@ -55,3 +55,43 @@ jobs:
       BuildConfiguration: Debug
       InstallPlatform: Linux-x64
   - template: azure-pipelines/templates/test-null-renderer.yml
+
+- job: format
+  displayName: 'Format'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  container: 'openblack/compile-farm-ci:linux-amd64-gcc'
+
+  steps:
+    - script: |
+        set -eo pipefail
+        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+        sudo add-apt-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-8 main"
+        sudo apt-get update
+        sudo apt-get install -y clang-format-8 clang-tidy-8
+        sudo update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-8 100
+      displayName: 'Install clang'
```

Comment:
Would be better off if the docker had this installed already. @handsomematt

Reply:
Angular has static builds of clang-format that get updated every so often: https://github.com/angular/clang-format

3-4MB binary is all you need. Keep in mind these are not directly tied to clang releases and _may_ produce slight variations in output in each version.

Reply:
Thanks. I'll try that. It does require npm though.

Reply:
Either @handsomematt has to install clang-format on the container or npm

Reply:
The way I was using it was simply to download the binary directly from github, mark as a+x, then execute. No npm necessary really.

Reply:
You can download whatever is on current master from this link: https://github.com/angular/clang-format/blob/master/bin/linux_x64/clang-format?raw=true